### PR TITLE
Respect react event handler

### PR DESCRIPTION
**Pull request by wclr**
`@cycle/react` currenlty replaces react handler on the element with its own event handler, this probably should be changed and the existing handler should be respected and executed.

The case for this: sometimes react expects that the event handler to return a value (for example if to pass the event or something like this). So this PR checks if handler exists and executes it after cycle event handler.

Also added prettier config. If ok with this PR, probably need to check and set new formatting of other files as well.

@staltz

### CODE REVIEW

File: src/Incorporator.ts [hunk 1]
```diff
@@ -32,7 +32,15 @@ export default class Incorporator extends PureComponent<Props, State> {
     const handlers = scope.getSelectorHandlers(this.selector);
     for (const evType of Object.keys(handlers)) {
       const onFoo = `on${evType[0].toUpperCase()}${evType.slice(1)}`;
-      props[onFoo] = (ev: any) => handlers[evType]._n(ev);
+      const oldHandler =
+        typeof props[onFoo] === 'function' ? props[onFoo] : undefined;
+      props[onFoo] = (ev: any) => {
```
Comment (staltz) [inline src/Incorporator.ts]: Can you put the `if` condition outside of this arrow function? In the default case where we use the Cycle.js style without callbacks (the most common case when using cycle/react), we shouldn't wrap the handler function in an unnecessary function.

**Review by staltz — CHANGES_REQUESTED** [review]
Good PR, I think this change makes sense. I have one request, written inline.

Note, ideally the addition of prettier would be a separate PR (or at least separate commit), it's not good for codebase maintenance (e.g. using git blame) to have unrelated changes all together in one commit, **but** I don't mind merging it like this right now. So, no action needed from you, I'm just informing what's preferable for the next times.

**Comment by wclr** [comment]
Made PR with formatting https://github.com/cyclejs/react/pull/14 and then tried to rebase this PR, not sure if I did right with this PR.

**Comment by staltz** [comment]
The `if` change looks good. Now we just need this PR rebased or conflicts resolved, to merge. If you have difficulties rebasing, you can also open another branch/PR and `git cherry-pick` the commit.

**Review by staltz — APPROVED** [review]

**Comment by staltz** [comment]
I suppose you're done and this is ready to merge? @whitecolor

**Comment by wclr** [comment]
I believe yes, if not count a mess with commits.

**Comment by staltz** [comment]
Thanks!

**Comment by staltz** [comment]
Release 2.9.0

**Pull request merged**